server/src/endpoint/claims/authorize-thirdParty.ts

```diff
@@ -0,0 +1,38 @@
+import Joi from "joi";
+import { RequestHandler } from "../request-handler-wrapper";
+import { validationBadRequest } from "../../utils/errors";
+import { validate, ValidationSchema } from "../../utils/validate";
+
+const claim = Joi.object().keys({
+  key: Joi.string().required(),
+  type: Joi.string().required(),
+  value: Joi.string().required(),
+  evidence: Joi.array().min(1)
+});
+
+const bodyValidation: ValidationSchema<server.ThirdPartyAuthorizeRequest> = {
+  claims: Joi.array().items(claim).min(1),
+  issuer: Joi.string().required()
+};
+
+const authorizeThirdParty: RequestHandler<
+  void,
+  void,
+  server.ThirdPartyAuthorizeRequest,
+  server.SuccessResponse | server.BadRequestGeneralResponse
+> = async ({ body, services }) => {
+  const bodyValidationResult = await validate(body, bodyValidation);
+
+  if (bodyValidationResult.isInvalid) {
+    return validationBadRequest(bodyValidationResult.errors);
+  }
+
+  if (body.issuer === "gpib") {
```

Comment:
This might have to change to a config or something later, but happy with it for the time being